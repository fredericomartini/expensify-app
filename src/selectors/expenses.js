import moment from 'moment';

// Visible Expenses
export default (expenses, {
  text, sortBy, startDate, endDate,
}) => expenses
  .filter((expense) => {
    const createdAtMoment = moment(expense.createdAt);
    const startDateMatch = startDate ? startDate.isSameOrBefore(createdAtMoment, 'day') : true;
    const endDateMatch = endDate ? endDate.isSameOrAfter(createdAtMoment, 'day') : true;
    const textMatch = expense.description.toLowerCase().includes(text.toLowerCase());
    return startDateMatch && endDateMatch && textMatch;
  })
  .sort((a, b) => {
    if (sortBy === 'date') {
      return moment(a.createdAt).isBefore(moment(b.createdAt)) || -1;
    }
    if (sortBy === 'amount') {
      return a.amount < b.amount || -1;
    }
    return [b, a];
  });
