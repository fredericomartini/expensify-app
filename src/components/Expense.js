import React from 'react';
import { connect } from 'react-redux';
import { removeExpense } from '../actions/expenses';

const Expense = ({
  dispatch, id, description, amount, createdAt,
}) => (
  <div>
    <h3>{description}</h3>
    <p>
      Amount:
      {' '}
      {amount}
      {' '}
createdAt:
      {createdAt}
    </p>
    <button
      onClick={() => {
        dispatch(removeExpense({ id }));
      }}
    >
      Remove
    </button>
  </div>
);

export default connect()(Expense);
