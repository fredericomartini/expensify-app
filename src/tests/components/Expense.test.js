import React from 'react';
import { shallow } from 'enzyme';
import Expense from '../../components/Expense';
import expenses from '../fixtures/expenses';

test('should RENDER ONE EXPENSE', () => {
  const wrapper = shallow(<Expense expense={expenses[0]} />);
  expect(wrapper).toMatchSnapshot();
});
