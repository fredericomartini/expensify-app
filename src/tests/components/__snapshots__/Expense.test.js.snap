// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should RENDER ONE EXPENSE 1`] = `
<div>
  <Link
    replace={false}
    to="/edit/1"
  >
    <h3>
      Watter bill
    </h3>
  </Link>
  <p>
    Amount:
     
    199
     
    createdAt:
    31/12/1969
  </p>
</div>
`;
